src/compiler/types.rs

```diff
@@ -1438,43 +1531,100 @@ impl DiagnosticRelatedInformationInterface for Diagnostic {
             }
         }
     }
+
+    fn message_text(&self) -> &DiagnosticMessageText {
+        match self {
+            Diagnostic::DiagnosticWithLocation(diagnostic_with_location) => {
+                diagnostic_with_location.message_text()
+            }
+            Diagnostic::DiagnosticWithDetachedLocation(diagnostic_with_detached_location) => {
+                diagnostic_with_detached_location.message_text()
+            }
+        }
+    }
 }
 
 impl DiagnosticInterface for Diagnostic {}
 
+#[derive(Clone, Debug)]
+pub enum DiagnosticMessageText {
```

Comment:
This corresponds to `message_text: string | DiagnosticMessageChain` in the Typescript diagnostic types